app/controllers/user_sessions_controller.rb

```diff
@@ -50,7 +50,14 @@ def handle_social_login_flow(auth)
         redirect_to return_to + hash_params, notice: "Already linked to another account!"
       end
     else # not signed in
-      if @identity&.user.present?
+      @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : @identity.user
+      if @user&.status&.zero?
+        flash[:error] = I18n.t('user_sessions_controller.user_has_been_banned', username: @user.username).html_safe
+        redirect_to root_url
+      elsif @user&.status == 5
+        flash[:error] = I18n.t('user_sessions_controller.user_has_been_moderated', username: @user.username).html_safe
+        redirect_to root_url
```

Comment:
waiting for redirect to be rectified first

Reply:
Alright I fixed that problem

Reply:
we will come back here after merging your other prs